### DISCUSSION TITLE
how to get faster http response.

### DISCUSSION THREAD

**Discussion by pbelx** [vlang/v#19449] [Questions and Answers]
I would like to get faster response times using the following code.
        
        import net.http
        import json
        import benchmark
        
        struct Reply{
	        album_id int
            id int
            title string
            url string
            thumbnail_url string
        
        }
        
        fn main(){
	        println("starting")
	        mut b := benchmark.start()
	        repl := http.get("https://jsonplaceholder.typicode.com/photos") or{
		        eprintln("failed to get")
		        return
	        }
	        b.measure('get http')
	        print(repl.body)
	        b.measure('print body done')
	        jdecoded := json.decode([]Reply,repl.body)or {
		        eprintln('Failed to parse json')
		        return
	        }
	        b.measure('decode json')
	        print(jdecoded)
	        b.measure('print json')
        
        
        }

currently the time for response is around 21 seconds. Any ideas on making it take less time? Thanks.

**Comment by JalonSolov** [comment]
I just tried on Linux, and got this (filtering to only the `SPENT` lines):
```
$ time v run http.v | grep SPENT
SPENT  219.873 ms in get http
SPENT    0.413 ms in print body done
SPENT    5.871 ms in decode json
SPENT    6.657 ms in print json

real    0m0.505s
user    0m0.279s
sys     0m0.146s
$
```
So... to compile the code then run it, it took about 1/2 second.

If you're running on Windows, sounds like the way to make it take less time is to switch to Linux.  :-)

**Comment by ttytm** [comment]
Averages on macOS with my current crappy wifi:
- using `net.http`
  `SPENT  562.916 ms in get http`

- using [`vibe`](https://github.com/ttytm/vibe)
  
  ```diff
  - import net.http
  + import vibe
  ...
  - repl := http.get("https://jsonplaceholder.typicode.com/photos") or{
  + repl := vibe.get("https://jsonplaceholder.typicode.com/photos") or{
  ```
  `SPENT  347.887 ms in get http`

**Reply by pbelx** [reply]
Hello. 
I get the following error when trying to run / compile the code using vibe.

       error: 'CURLE_SETOPT_OPTION_SYNTAX' undeclared

**Reply by ttytm** [reply]
Uff, I forgot that windows needs quite some manual work to get it running atm. Gonna try to extend windows instructions and improve out of the box windows support soon. Windows isn't too shitty anymore, so I like to support it better :+1:. Sorry about this.

**Reply by pbelx** [reply]
Forgot to mention.I am running on Debian 10.

**Reply by ttytm** [reply]
Okay awesome. What your curl version?

**Reply by pbelx** [reply]
curl --version

    curl 7.64.0 (x86_64-pc-linux-gnu) libcurl/7.64.0 OpenSSL/1.1.1n zlib/1.2.11 libidn2/2.0.5 libpsl/0.20.2 (+libidn2/2.0.5) libssh2/1.8.0 nghttp2/1.36.0 librtmp/2.3
          Release-Date: 2019-02-06

**Reply by ttytm** [reply]
Thanks. Currently, the minimum version is the one that is available for latest ubuntu lts - 7.81. Gonna look into extending the version support.

**Reply by ttytm** [reply]
@pbelx 

It now works with curl versions as low as 7.64. The minimum required version will be raised after debian 10 EOL - likely to 7.74 used in deb 11.
